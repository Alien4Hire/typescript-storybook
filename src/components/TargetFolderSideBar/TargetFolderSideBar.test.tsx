import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { ThemeProvider, StyledEngineProvider } from '@mui/material/styles';
import { MemoryRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';

import { theme } from '../../theme';
import '../../setupTests';
import { setupStore } from '../../app/store';
import TargetFolderSideBar from './TargetFolderSideBar';

beforeAll(() => {
  window.getComputedStyle = jest.fn();
});

beforeEach((): void => {
  fetchMock.resetMocks();
});

const sampleFolders = [
  {
    id: '15',
    name: 'Instructor Guide & FAQ',
    type: 'folder' as 'folder',
    children: [
      {
        id: '16',
        name: 'Intro quiz',
        type: 'folder' as 'folder',
        children: [],
      },
    ],
  },
  {
    id: '18',
    name: 'Intro Survey',
    type: 'folder' as 'folder',
    children: [],
  },
  {
    id: '3',
    name: 'Checkpoint Survey #1',
    type: 'folder' as 'folder',
    children: [],
  },
];

const baseProps = {
  isOpen: true,
  folders: [...sampleFolders],
};

const renderComponent = () => {
  const store = setupStore();

  return render(
    <Provider store={store}>
      <Router>
        <StyledEngineProvider injectFirst>
          <ThemeProvider theme={theme}>
            <TargetFolderSideBar {...baseProps} />
          </ThemeProvider>
        </StyledEngineProvider>
      </Router>
    </Provider>
  );
};

describe('<TargetFolderSideBar />', () => {
  describe('Should render properly', () => {
    it('renders without crashing', () => {
      const view = renderComponent();
      expect(view).toMatchSnapshot();
    });
    it('should render Move button', () => {
      renderComponent();
      const buttonMove = screen.getByTestId('move-button');
      expect(buttonMove).toBeInTheDocument();
    });
    it('should display a list of folder items', () => {
      renderComponent();
      const folderList = screen.getByTestId('folder-list');
      const itemRootLevel = screen.getByText('Root Level');
      const checkedItem = screen.queryByTestId('item-selected-checkmark');
      expect(folderList).toBeInTheDocument();
      expect(itemRootLevel).toBeInTheDocument();
      expect(checkedItem).not.toBeInTheDocument();
    });
  });
  describe('clicking on folder item', () => {
    it('should display a check mark', () => {
      renderComponent();
      const folderList = screen.getByTestId('folder-list');
      const itemRootLevel = screen.getByText('Root Level');
      fireEvent.click(itemRootLevel);
      const checkedItem = screen.queryByTestId('item-selected-checkmark');
      expect(folderList).toBeInTheDocument();
      expect(itemRootLevel).toBeInTheDocument();
      expect(checkedItem).toBeInTheDocument();
    });
  });
  describe('displaying nested folder items', () => {
    it('should display a left chevron icon if nested folder', () => {
      renderComponent();
      const folderList = screen.getByTestId('folder-list');
      const chevronIcon = screen.getByTestId('folder-expanded-icon');
      expect(folderList).toBeInTheDocument();
      expect(chevronIcon).toBeInTheDocument();
    });
    it('should be visible nested folder on chevron click', () => {
      renderComponent();
      const folderList = screen.getByTestId('folder-list');
      const chevronIcon = screen.getByTestId('folder-expanded-icon');
      fireEvent.click(chevronIcon);
      const IntroQuizNested = screen.queryByText(/Intro quiz/);
      expect(folderList).toBeInTheDocument();
      expect(IntroQuizNested).toBeInTheDocument();
    });
  });
});
