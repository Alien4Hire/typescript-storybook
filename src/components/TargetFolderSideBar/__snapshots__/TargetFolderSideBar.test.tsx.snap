// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TargetFolderSideBar /> Should render properly renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c15 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  font-size: 1.5rem;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.c16 {
  height: 0;
  overflow: hidden;
  -webkit-transition: height 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: height 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  visibility: hidden;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.c18 {
  width: 100%;
}

.c4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c4::-moz-focus-inner {
  border-style: none;
}

.c4.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

.c3 {
  margin: 0;
  color: #080808;
  font-size: 18px;
  font-family: Source Sans Pro;
  font-weight: 600;
  line-height: 1.167;
}

.c12 {
  margin: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen,Ubuntu,Cantarell,"Fira Sans","Droid Sans","Helvetica Neue",Source Sans Pro,sans-serif;
  font-weight: normal;
  font-size: 1rem;
  line-height: 1.5;
  display: block;
}

.c5 {
  font-size: 14px;
  text-transform: uppercase;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen,Ubuntu,Cantarell,"Fira Sans","Droid Sans","Helvetica Neue",Source Sans Pro,sans-serif;
  font-weight: normal;
  line-height: 1.75;
  min-width: 64px;
  padding: 6px 16px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,box-shadow 250ms cubic-bezier(0.4,0,0.2,1) 0ms,border-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,color 250ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,box-shadow 250ms cubic-bezier(0.4,0,0.2,1) 0ms,border-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,color 250ms cubic-bezier(0.4,0,0.2,1) 0ms;
  color: #fff;
  background-color: #00758e;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
}

.c5:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgb(0,81,99);
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

.c5:active {
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
}

.c5.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.c5.Mui-disabled {
  color: rgba(0,0,0,0.26);
  box-shadow: none;
  background-color: rgba(0,0,0,0.12);
}

.c6 {
  list-style: none;
  margin: 0;
  padding: 0;
  position: relative;
  padding-top: 8px;
  padding-bottom: 8px;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  box-sizing: border-box;
  text-align: left;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
}

.c8.Mui-focusVisible {
  background-color: rgba(0,0,0,0.12);
}

.c8.Mui-selected {
  background-color: rgba(0,117,142,0.08);
}

.c8.Mui-selected.Mui-focusVisible {
  background-color: rgba(0,117,142,0.2);
}

.c8.Mui-disabled {
  opacity: 0.38;
}

.c10 {
  min-width: 56px;
  color: rgba(0,0,0,0.54);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  min-width: 30px;
}

.c13 {
  min-width: 56px;
  color: rgba(0,0,0,0.54);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c11 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  min-width: 0;
  margin-top: 4px;
  margin-bottom: 4px;
}

.c0 {
  margin: 180px 12px 12px 12px;
  text-transform: none;
  height: 100%;
  margin-bottom: 12px;
}

.c1 {
  background-color: #ffffff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 100%;
  width: 400px;
  border: 1px solid #c8c8c8;
}

.c2 {
  padding: 20px 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-bottom: 1px solid #c8c8c8;
}

.c7 {
  overflow: auto;
  height: 100%;
  padding-bottom: 5px;
}

.c14 {
  background-color: #ffffff;
  border-bottom: 1px solid #c8c8c8;
  padding: 8px;
}

.c9 {
  background-color: #ffffff;
  border-bottom: 1px solid #c8c8c8;
  border-top: 1px solid #c8c8c8;
  padding: 8px;
}

@media print {
  .c4 {
    color-adjust: exact;
  }
}

@media (hover:none) {
  .c5:hover {
    background-color: #00758e;
  }
}

<body>
    <div>
      <div
        class="c0 MuiBox-root"
      >
        <div
          class="c1 MuiBox-root"
        >
          <div
            class="c2 MuiBox-root"
          >
            <h3
              class="c3 MuiTypography-root MuiTypography-h3"
            >
              Select a Destination Folder
            </h3>
            <button
              class="c4 MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled c5"
              data-testid="move-button"
              disabled=""
              tabindex="-1"
              test-id="move-button"
              type="button"
            >
              Move
            </button>
          </div>
          <div
            class="MuiBox-root"
          >
            <ul
              class="c6 MuiList-root MuiList-padding c7"
              data-testid="folder-list"
            >
              <li
                class="c8 MuiListItem-root MuiListItem-gutters MuiListItem-padding c9"
              >
                <div
                  class="c10 MuiListItemIcon-root"
                />
                <div
                  class="c11 MuiListItemText-root"
                >
                  <span
                    class="c12 MuiTypography-root MuiTypography-body1 MuiListItemText-primary"
                  >
                    Root Level
                  </span>
                </div>
                <div
                  class="c13 MuiListItemIcon-root"
                />
              </li>
              <div
                class="MuiBox-root"
              >
                <li
                  class="c8 MuiListItem-root MuiListItem-gutters MuiListItem-padding c14"
                >
                  <div
                    class="c10 MuiListItemIcon-root"
                    data-testid="folder-expanded-icon-container"
                  >
                    <svg
                      aria-hidden="true"
                      class="c15 MuiSvgIcon-root MuiSvgIcon-fontSizeMedium"
                      data-testid="folder-expanded-icon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                      />
                    </svg>
                  </div>
                  <div
                    class="c11 MuiListItemText-root"
                  >
                    <span
                      class="c12 MuiTypography-root MuiTypography-body1 MuiListItemText-primary"
                    >
                      Instructor Guide & FAQ
                    </span>
                  </div>
                  <div
                    class="c13 MuiListItemIcon-root"
                  />
                </li>
                <ul
                  class="c16 MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden"
                  style="min-height: 0px;"
                >
                  <div
                    class="c17 MuiCollapse-wrapper MuiCollapse-vertical"
                  >
                    <div
                      class="c18 MuiCollapse-wrapperInner MuiCollapse-vertical"
                    >
                      <div
                        class="MuiBox-root"
                      >
                        <li
                          class="c8 MuiListItem-root MuiListItem-gutters MuiListItem-padding c14"
                        >
                          <div
                            class="c10 MuiListItemIcon-root"
                            data-testid="folder-expanded-icon-container"
                          />
                          <div
                            class="c11 MuiListItemText-root"
                          >
                            <span
                              class="c12 MuiTypography-root MuiTypography-body1 MuiListItemText-primary"
                            >
                              Intro quiz
                            </span>
                          </div>
                          <div
                            class="c13 MuiListItemIcon-root"
                          />
                        </li>
                        <ul
                          class="c16 MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden"
                          style="min-height: 0px;"
                        >
                          <div
                            class="c17 MuiCollapse-wrapper MuiCollapse-vertical"
                          >
                            <div
                              class="c18 MuiCollapse-wrapperInner MuiCollapse-vertical"
                            />
                          </div>
                        </ul>
                      </div>
                    </div>
                  </div>
                </ul>
              </div>
              <div
                class="MuiBox-root"
              >
                <li
                  class="c8 MuiListItem-root MuiListItem-gutters MuiListItem-padding c14"
                >
                  <div
                    class="c10 MuiListItemIcon-root"
                    data-testid="folder-expanded-icon-container"
                  />
                  <div
                    class="c11 MuiListItemText-root"
                  >
                    <span
                      class="c12 MuiTypography-root MuiTypography-body1 MuiListItemText-primary"
                    >
                      Intro Survey
                    </span>
                  </div>
                  <div
                    class="c13 MuiListItemIcon-root"
                  />
                </li>
                <ul
                  class="c16 MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden"
                  style="min-height: 0px;"
                >
                  <div
                    class="c17 MuiCollapse-wrapper MuiCollapse-vertical"
                  >
                    <div
                      class="c18 MuiCollapse-wrapperInner MuiCollapse-vertical"
                    />
                  </div>
                </ul>
              </div>
              <div
                class="MuiBox-root"
              >
                <li
                  class="c8 MuiListItem-root MuiListItem-gutters MuiListItem-padding c14"
                >
                  <div
                    class="c10 MuiListItemIcon-root"
                    data-testid="folder-expanded-icon-container"
                  />
                  <div
                    class="c11 MuiListItemText-root"
                  >
                    <span
                      class="c12 MuiTypography-root MuiTypography-body1 MuiListItemText-primary"
                    >
                      Checkpoint Survey #1
                    </span>
                  </div>
                  <div
                    class="c13 MuiListItemIcon-root"
                  />
                </li>
                <ul
                  class="c16 MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden"
                  style="min-height: 0px;"
                >
                  <div
                    class="c17 MuiCollapse-wrapper MuiCollapse-vertical"
                  >
                    <div
                      class="c18 MuiCollapse-wrapperInner MuiCollapse-vertical"
                    />
                  </div>
                </ul>
              </div>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": .c15 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  font-size: 1.5rem;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.c16 {
  height: 0;
  overflow: hidden;
  -webkit-transition: height 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: height 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  visibility: hidden;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.c18 {
  width: 100%;
}

.c4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c4::-moz-focus-inner {
  border-style: none;
}

.c4.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

.c3 {
  margin: 0;
  color: #080808;
  font-size: 18px;
  font-family: Source Sans Pro;
  font-weight: 600;
  line-height: 1.167;
}

.c12 {
  margin: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen,Ubuntu,Cantarell,"Fira Sans","Droid Sans","Helvetica Neue",Source Sans Pro,sans-serif;
  font-weight: normal;
  font-size: 1rem;
  line-height: 1.5;
  display: block;
}

.c5 {
  font-size: 14px;
  text-transform: uppercase;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen,Ubuntu,Cantarell,"Fira Sans","Droid Sans","Helvetica Neue",Source Sans Pro,sans-serif;
  font-weight: normal;
  line-height: 1.75;
  min-width: 64px;
  padding: 6px 16px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,box-shadow 250ms cubic-bezier(0.4,0,0.2,1) 0ms,border-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,color 250ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,box-shadow 250ms cubic-bezier(0.4,0,0.2,1) 0ms,border-color 250ms cubic-bezier(0.4,0,0.2,1) 0ms,color 250ms cubic-bezier(0.4,0,0.2,1) 0ms;
  color: #fff;
  background-color: #00758e;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
}

.c5:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgb(0,81,99);
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

.c5:active {
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
}

.c5.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.c5.Mui-disabled {
  color: rgba(0,0,0,0.26);
  box-shadow: none;
  background-color: rgba(0,0,0,0.12);
}

.c6 {
  list-style: none;
  margin: 0;
  padding: 0;
  position: relative;
  padding-top: 8px;
  padding-bottom: 8px;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  box-sizing: border-box;
  text-align: left;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
}

.c8.Mui-focusVisible {
  background-color: rgba(0,0,0,0.12);
}

.c8.Mui-selected {
  background-color: rgba(0,117,142,0.08);
}

.c8.Mui-selected.Mui-focusVisible {
  background-color: rgba(0,117,142,0.2);
}

.c8.Mui-disabled {
  opacity: 0.38;
}

.c10 {
  min-width: 56px;
  color: rgba(0,0,0,0.54);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  min-width: 30px;
}

.c13 {
  min-width: 56px;
  color: rgba(0,0,0,0.54);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c11 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  min-width: 0;
  margin-top: 4px;
  margin-bottom: 4px;
}

.c0 {
  margin: 180px 12px 12px 12px;
  text-transform: none;
  height: 100%;
  margin-bottom: 12px;
}

.c1 {
  background-color: #ffffff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 100%;
  width: 400px;
  border: 1px solid #c8c8c8;
}

.c2 {
  padding: 20px 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-bottom: 1px solid #c8c8c8;
}

.c7 {
  overflow: auto;
  height: 100%;
  padding-bottom: 5px;
}

.c14 {
  background-color: #ffffff;
  border-bottom: 1px solid #c8c8c8;
  padding: 8px;
}

.c9 {
  background-color: #ffffff;
  border-bottom: 1px solid #c8c8c8;
  border-top: 1px solid #c8c8c8;
  padding: 8px;
}

@media print {
  .c4 {
    color-adjust: exact;
  }
}

@media (hover:none) {
  .c5:hover {
    background-color: #00758e;
  }
}

<div>
    <div
      class="c0 MuiBox-root"
    >
      <div
        class="c1 MuiBox-root"
      >
        <div
          class="c2 MuiBox-root"
        >
          <h3
            class="c3 MuiTypography-root MuiTypography-h3"
          >
            Select a Destination Folder
          </h3>
          <button
            class="c4 MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled c5"
            data-testid="move-button"
            disabled=""
            tabindex="-1"
            test-id="move-button"
            type="button"
          >
            Move
          </button>
        </div>
        <div
          class="MuiBox-root"
        >
          <ul
            class="c6 MuiList-root MuiList-padding c7"
            data-testid="folder-list"
          >
            <li
              class="c8 MuiListItem-root MuiListItem-gutters MuiListItem-padding c9"
            >
              <div
                class="c10 MuiListItemIcon-root"
              />
              <div
                class="c11 MuiListItemText-root"
              >
                <span
                  class="c12 MuiTypography-root MuiTypography-body1 MuiListItemText-primary"
                >
                  Root Level
                </span>
              </div>
              <div
                class="c13 MuiListItemIcon-root"
              />
            </li>
            <div
              class="MuiBox-root"
            >
              <li
                class="c8 MuiListItem-root MuiListItem-gutters MuiListItem-padding c14"
              >
                <div
                  class="c10 MuiListItemIcon-root"
                  data-testid="folder-expanded-icon-container"
                >
                  <svg
                    aria-hidden="true"
                    class="c15 MuiSvgIcon-root MuiSvgIcon-fontSizeMedium"
                    data-testid="folder-expanded-icon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                    />
                  </svg>
                </div>
                <div
                  class="c11 MuiListItemText-root"
                >
                  <span
                    class="c12 MuiTypography-root MuiTypography-body1 MuiListItemText-primary"
                  >
                    Instructor Guide & FAQ
                  </span>
                </div>
                <div
                  class="c13 MuiListItemIcon-root"
                />
              </li>
              <ul
                class="c16 MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden"
                style="min-height: 0px;"
              >
                <div
                  class="c17 MuiCollapse-wrapper MuiCollapse-vertical"
                >
                  <div
                    class="c18 MuiCollapse-wrapperInner MuiCollapse-vertical"
                  >
                    <div
                      class="MuiBox-root"
                    >
                      <li
                        class="c8 MuiListItem-root MuiListItem-gutters MuiListItem-padding c14"
                      >
                        <div
                          class="c10 MuiListItemIcon-root"
                          data-testid="folder-expanded-icon-container"
                        />
                        <div
                          class="c11 MuiListItemText-root"
                        >
                          <span
                            class="c12 MuiTypography-root MuiTypography-body1 MuiListItemText-primary"
                          >
                            Intro quiz
                          </span>
                        </div>
                        <div
                          class="c13 MuiListItemIcon-root"
                        />
                      </li>
                      <ul
                        class="c16 MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden"
                        style="min-height: 0px;"
                      >
                        <div
                          class="c17 MuiCollapse-wrapper MuiCollapse-vertical"
                        >
                          <div
                            class="c18 MuiCollapse-wrapperInner MuiCollapse-vertical"
                          />
                        </div>
                      </ul>
                    </div>
                  </div>
                </div>
              </ul>
            </div>
            <div
              class="MuiBox-root"
            >
              <li
                class="c8 MuiListItem-root MuiListItem-gutters MuiListItem-padding c14"
              >
                <div
                  class="c10 MuiListItemIcon-root"
                  data-testid="folder-expanded-icon-container"
                />
                <div
                  class="c11 MuiListItemText-root"
                >
                  <span
                    class="c12 MuiTypography-root MuiTypography-body1 MuiListItemText-primary"
                  >
                    Intro Survey
                  </span>
                </div>
                <div
                  class="c13 MuiListItemIcon-root"
                />
              </li>
              <ul
                class="c16 MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden"
                style="min-height: 0px;"
              >
                <div
                  class="c17 MuiCollapse-wrapper MuiCollapse-vertical"
                >
                  <div
                    class="c18 MuiCollapse-wrapperInner MuiCollapse-vertical"
                  />
                </div>
              </ul>
            </div>
            <div
              class="MuiBox-root"
            >
              <li
                class="c8 MuiListItem-root MuiListItem-gutters MuiListItem-padding c14"
              >
                <div
                  class="c10 MuiListItemIcon-root"
                  data-testid="folder-expanded-icon-container"
                />
                <div
                  class="c11 MuiListItemText-root"
                >
                  <span
                    class="c12 MuiTypography-root MuiTypography-body1 MuiListItemText-primary"
                  >
                    Checkpoint Survey #1
                  </span>
                </div>
                <div
                  class="c13 MuiListItemIcon-root"
                />
              </li>
              <ul
                class="c16 MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden"
                style="min-height: 0px;"
              >
                <div
                  class="c17 MuiCollapse-wrapper MuiCollapse-vertical"
                >
                  <div
                    class="c18 MuiCollapse-wrapperInner MuiCollapse-vertical"
                  />
                </div>
              </ul>
            </div>
          </ul>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
